### DISCUSSION TITLE
Report a bug.

### DISCUSSION THREAD

**Discussion by lmmqxyx404** [rbatis/rbatis#312] [General]
```
extern crate rbatis;

use rbdc_mysql::driver::MysqlDriver;
#[tokio::main]
async fn main() {
    // println!("Hello, world!");
    let rb = rbatis::Rbatis::new();
    let res = rb.init(MysqlDriver {}, "mysql://root:admin@localhost:13308/test");
    if res.is_ok() {
        println!("Hello, world!");
    } else {
        println!("failed");
    }
}
```
The database_url is wrong. but the code could also run.
There may be a bug with the function  ```init()```

**Comment by zhuxiujia** [comment]
> ```
> extern crate rbatis;
> 
> use rbdc_mysql::driver::MysqlDriver;
> #[tokio::main]
> async fn main() {
>     // println!("Hello, world!");
>     let rb = rbatis::Rbatis::new();
>     let res = rb.init(MysqlDriver {}, "mysql://root:admin@localhost:13308/test");
>     if res.is_ok() {
>         println!("Hello, world!");
>     } else {
>         println!("failed");
>     }
> }
> ```
> 
> The database_url is wrong. but the code could also run. There may be a bug with the function `init()`

you must use ```rb.link()``` replace ``` rb.init() ```
check database url  must call method init method and than call  ```rb.try_acquire().await?;```

**Reply by lmmqxyx404** [reply]
Thanks a lot.Your suggestion works well.
But in my opinion, such a design is incomprehensible.
The  intuitive understanding of ```init()``` should be a ```Result(successful ,failed)```.
I do sincerely feel that the previous design(``` link()```) is better.
